katdal/chunkstore.py

```diff
@@ -437,6 +441,58 @@ def put_dask_array(self, array_name, array, offset=()):
         out_chunks = tuple(len(c) * (1,) for c in array.chunks)
         return da.Array(dask_graph, out_name, out_chunks, np.object)
 
+    def list_chunk_ids(self, array_name):
+        """List all chunk ID strings associated with given array in chunk store.
+
+        Parameters
+        ----------
+        array_name : string
+            Identifier of array in chunk store
+
+        Returns
+        -------
+        chunk_ids : list of string
+            List of chunk identifier strings (e.g. '00012_01024_00000')
```

Comment:
Maybe document the `NotImplementedError` case. By the way, have you thought about doing `return NotImplemented`, which I think is what Python uses for unimplemented operator overloads? It would probably take more code though (since you'd have to explicitly handle it in has_array, rather than letting an exception bubble out).

Reply:
I saw in tab completion that there are two NotImplemented's and wondered about it...

The return version seems to be specialised to the operator case where the comparison can happen thousands of times, and result in more code as you mention.

I flirted with the idea of `has_array` falling back to `has_dask_array().compute()` if `list_chunk_ids` is not there, but then it all gets quite cyclical.